nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -656,10 +666,23 @@ impl<'a> Planner<'a> {
         // decommissioned.
         let mut num_existing_kind_zones = 0;
         for sled_id in self.input.all_sled_ids(SledFilter::InService) {
-            let num_zones_of_kind = self
+            let zone_kind = ZoneKind::from(zone_kind);
+
+            // Internal DNS is special: if we have an expunged internal DNS zone
```

Comment:
This is the actual bugfix: don't just count `in-service` for internal DNS.

Reply:
There seems to be some sort of conservation-of-specialness for internal DNS: you can move this check around, but it can't be shared with other services or destroyed.